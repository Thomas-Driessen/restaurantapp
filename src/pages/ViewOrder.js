import React from 'react';
import NavBar from '../components/Navigation bars/NavBar';
import PreviousOrders from '../components/Previous Orders List/PreviousOrders'
import CurrentOrder from '../components/Current Order/CurrentOrder'
import { Button } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import currentFoodList from '../components/Current Order/CurrentFoodList'
import currentDrinkList from '../components/Current Order/CurrentDrinkList'
import tableId from '../components/TableId'

class ViewOrder extends React.Component {
  constructor() {
    super();
    this.state = { tableNumber: 1,
      updatePastOrders: 0
    };
  }

  sendOrder(){
    let order = [];
    currentDrinkList.map(currentDrink => {
        var drink ={
            tableId: tableId,
            paid: false,
            drinkId: currentDrink.id
        };
        let item = {
            title: currentDrink.title
        }
        order.push(item);
    fetch('/api/orderdrink', {
        method: 'POST',
        mode: 'cors',
        headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
    },
    body: JSON.stringify(drink)
    })
    return "Succes";
    })

    currentFoodList.map(currentFood => {
        var food ={
            tableId: tableId,
            paid: false,
            foodId: currentFood.id
        };
        let item = {
            title: currentFood.title
        }
        order.push(item);
    fetch('/api/orderfood', {
        method: 'POST',
        mode: 'cors',
        headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
    },
    body: JSON.stringify(food)
    })
    return "Succes";
    })

    currentDrinkList.splice(0,currentDrinkList.length);
    currentFoodList.splice(0,currentFoodList.length);
    sessionStorage.removeItem("currentDrinkList");
    sessionStorage.removeItem("currentFoodList");
    
    fetch('/sendorder', {
        method: 'POST',
        mode: 'cors',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(order)
    })
    
    this.setState({ previousOrderNumbers: this.state.previousOrderNumbers + 1 })  
  }

  render(){
  return(
      <div>
          <NavBar/>
          <Grid container spacing={0} style={{padding: 25}}>
            <Grid item xs={12} sm={6} lg={4} xl={3}>
                <Button color="secondary"variant="outlined">
                  Pay for orders
                </Button>
              <CurrentOrder sendOrder={ () => this.sendOrder()}/>
              <PreviousOrders previousOrderNumbers={this.state.previousOrderNumbers}/>
            </Grid>
          </Grid>
      </div>
  )};
}
export default ViewOrder