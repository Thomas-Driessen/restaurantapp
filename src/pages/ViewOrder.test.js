import React from 'react';
import { render } from '@testing-library/react';
import ViewOrder from './ViewOrder'
import renderer from 'react-test-renderer';

it('ViewOrder renders correctly', () => {
  const tree = renderer
    .create(<ViewOrder />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render ViewOrder', () => {
  const { getByText } = render(<ViewOrder/>);
  const linkElement = getByText(/Pay for orders/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render ViewOrder', () => {
  const { getByText } = render(<ViewOrder/>);
  const linkElement = getByText(/This is your current order/i);
  expect(linkElement).toBeInTheDocument();
});