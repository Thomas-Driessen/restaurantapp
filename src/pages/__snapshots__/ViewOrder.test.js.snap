// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewOrder renders correctly 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          text-align="center"
        >
          Restaurant
        </h6>
        <div
          className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
        >
          <div
            style={
              Object {
                "float": "right",
              }
            }
          >
            <div
              aria-label="text primary button group"
              className="MuiButtonGroup-root"
              role="group"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Menu
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiButton-startIcon MuiButton-iconSizeLarge"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="3.2"
                      />
                      <path
                        d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                      />
                    </svg>
                  </span>
                  Scan QR
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
    style={
      Object {
        "padding": 25,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <div>
        <h2>
          This is your current order
        </h2>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          style={
            Object {
              "padding": 15,
            }
          }
        />
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          style={
            Object {
              "padding": 15,
            }
          }
        />
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Pay for orders
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <div>
        <h2>
          These are your previous orders
        </h2>
        <div
          className="MuiGrid-root MuiGrid-container"
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-start",
              "margin": "0 0 15px 0",
            }
          }
        >
          <b
            style={
              Object {
                "margin": "0 15px 0 0",
              }
            }
          >
            Total paid
          </b>
          <div
            id="totalPaid"
          >
            loading...
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-grid-sm-12"
          >
            <h3>
              Foods
            </h3>
          </div>
          <div
            className="MuiGrid-root MuiGrid-grid-sm-12"
          >
            <h3>
              Drinks
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
